### ISSUE TITLE
Provide automatic build arguments for Microbadger

### ISSUE THREAD

**Issue by ludovicc** [harbur/captain#61]
Microbadger uses some build-time arguments to document a Docker image.

Those are BUILD_DATE and VCS_REF. As they are defined from the environment, it's currently not possible to include them in the build args. Could those values be defined by default in captain?

See https://microbadger.com/#/labels for reference.

Thanks.

**Comment by spiddy** [comment]
As far as I understand the build-args needs to exist inside the Dockerfile to be compiled inside the image (and then those args are going to be available _inside_ the containers)

I haven't investigated yet, so correct me if I'm wrong:

- If we add build-args and those do not exist inside the Dockerfile, the build will fail (which means the build-args should be opt-in, not on by default)

As an alternative, I'd like to propose to use `--label` instead of `--build-args`:

- Labels would work for all Dockerfiles, not just the ones prepared for this feature.
- Those labels are _not_ going to be visible inside the Docker image (which would be a good feature to manage though)
- image labels should probably be namespaced to avoid conflicts with other tools (so instead of build.date, it would probably need to be `captain.build.date` (or something equivalent).

**Comment by ludovicc** [comment]
You are right, build-args not consumed will fail the build. I have tested that with captain.

Docker build supports adding labels and that looks like a nice feature to use here.

But instead of reinventing a set of labels specific to captain, why not use those defined by http://label-schema.org/ ? It's a first start at defining a standard to tag useful information on Docker images.

I would propose a specific extension to captain.yml  here:

hello-world:
  build: Dockerfile
  image: harbur/hello-world
  label_schema:
    name: “myname”
    description: “This service does awesome Things with Things”
    usage: “/usr/doc/app-usage.txt”
    url: ”http://hello.world”
    vcs-url: "https://github.com/hello/world"
    vendor: “Stark Industries”
    version: “1.2.3”
    depends-on: “tcp://mongo:8000”
    # added by default if label_schema is used:
    #   schema-version: "1.0"
    #   build-date=”2016-04-12T23:20:50.52Z”
    # added by default if git is detected and vcs-url is set:
    #   vcs-type = “git”
    #   vcs-ref = “279FA63”
